web/packages/teleport/src/Bots/EditBot.test.tsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { render, screen, userEvent } from 'design/utils/testing';
+
+import { EditBot } from 'teleport/Bots/EditBot';
+import { EditBotProps } from 'teleport/Bots/types';
+
+const makeProps = (): EditBotProps => ({
+  allRoles: [],
+  attempt: { status: '' },
+  name: 'bot-007',
+  onClose: () => {},
+  onEdit: () => {},
+  selectedRoles: [],
+  setSelectedRoles: () => {},
+});
```

Comment:
How about
```suggestion
const makeProps = (overrides: Partial<EditBotProps> = {}): EditBotProps => ({
  allRoles: [],
  attempt: { status: '' },
  name: 'bot-007',
  onClose: () => {},
  onEdit: () => {},
  selectedRoles: [],
  setSelectedRoles: () => {},
  ...overrides,
});
```

So you can simplify

```typescript
const props = makeProps();
props.selectedRoles = ['foo-role'];
```

to

```typescript
const props = makeProps({
  selectedRoles: ['foo-role'],
});
```